### PR TITLE
Feat: update BLE box connection process

### LINKED ISSUE #93: [Chore]: Change error appearence when recording without a linked box
## Summary

Once I start to record my trip without having a linked box with opensensemap, an error is shown. This error should better be shown as a warning

## Tasks

- [ ] Differentiate user facing errors as errors or warnings
- [ ] (Maybe introduce error levels, e.g. error, warn, info, ...)
- [ ] Decide on how to visualize warnings and errors

## Additional Context

While recording without being registered is not intended for our campaigns, some users might prefer this. Also, we have been using this approach in some school workshops

### LINKED ISSUE #110: [Bug]: after failed reconnect, when i connect to box i see recording button
%subj%

## Steps to Reproduce

1.
2.
3.

## Environment

*   Platform: [e.g., Android 15]
*   App Build Number: 
*   Other:

### LINKED ISSUE #111: [Bug]: when box disconnects due to powering off, it is not reflected in UI
## Steps to Reproduce

1.
2.
3.

## Environment

*   Platform: [e.g., Android 15]
*   App Build Number: 
*   Other:

### LINKED ISSUE #131: [Feature]: Rework logic of connecting to the senseBox
Currently it's checked if distance sensor characteristics stream is available. But if for some reason sensor is not available, then box can never connect.

## Describe the solution you'd like

Make sure that new data starts coming in through the characteristics streams. Per current senseBox code, sensors start sharing new data only after box is fully connected.

## Describe alternatives you've considered

Check if X% of sensors are available, but that doesn't eliminate issue with failing sensors and making sure that the box is fully loaded.

## Additional context

### PR DESCRIPTION
**Pull request by mashazyu**
Closes #131 
Closes #110 
Closes #111 (possibly)
Closes #93 

Rework ble device connection process

### Detailed Changes

*   Improve algorythm that verifies that ble device is properly connected (checks if data comes in through the first characteristics stream)
*   Cleanup blebloc (remove unused code and notifiers)
*   Update screen, when no devices is found to look like the screen when no tracks are found (unify ui)
*   Cleanup track recording/ui, when ble device disconnects (by user)
*.   Add tests

#### Test Suite 1: BLE Connection Process
1.1 Initial Connection Test
Objective: Verify improved connection verification algorithm
Steps:
- Open the app and navigate to home screen
- Ensure no device is currently connected
- Tap "Connect" button
- Select a senseBox device from the list
Expected Result:
- Connection process starts with proper loading state
- Connection verification checks characteristic streams for data
- Device connects successfully and sensor data appears
- UI shows connected state with start/stop recording buttons

Maria: Done with ATRAI box.
@felixerdy , can you please try to connect from you end both Atrai and older version(s) of boxes. Maybe a box with known broken sensors?

#### Test Suite 2: Error Handling and Recovery
2.1 Permanent Connection Failure Test
Objective: Verify improved handling of permanent BLE connection failures
Steps:
- Connect to a senseBox device
- Power off the senseBox device (simulate permanent failure)
- Wait for reconnection attempts to complete
Expected Result:
- App attempts reconnection up to max attempts (10)
- After max attempts, shows permanent connection error banner
- Error banner has proper spacing (16px above/below)
- Error banner shows with rounded corners and error styling
- No ongoing reconnection attempts in debug console

Steps:
- Hit X to close error message
Expected Result:
- Close button properly dismisses the error
- After dismissal, no sensor area is shown (empty state)

Maria: Done

#### Test Suite 3: Edge Cases and Stress Testing
3.1 Rapid Connection/Disconnection Test
Objective: Verify app handles rapid state changes gracefully
Steps:
- Connect to device
- Immediately disconnect
- Immediately try to connect again
Expected Result:
- App handles rapid state changes without crashes
- State is properly managed between transitions
- No memory leaks or stuck states
Success Criteria: App remains stable and responsive during rapid state changes

Maria: Done

3.2 Multiple Connection Attempts Test
Objective: Verify app can handle multiple connection attempts after failures
Steps:
- Fail to connect to a powered-off device
- Dismiss error banner
- Power on the device
- Try to connect again
Expected Result:
- App allows new connection attempts after error dismissal
- Previous error state doesn't interfere with new connections
- Connection process works normally
Success Criteria: App recovers from errors and allows new connection attempts

Maria: Done

3.3 Bluetooth State Changes Test
Objective: Verify app handles Bluetooth enable/disable gracefully
Steps:
- Connect to a device
- Disable Bluetooth from system settings
- Re-enable Bluetooth
Expected Result:
- App detects Bluetooth state changes
- Connection state is properly updated
- UI reflects current Bluetooth availability
Success Criteria: App properly handles Bluetooth state changes

Maria: Done

#### Test Suite 4: Integration and Performance
4.1 Track Recording Integration Test
Objective: Verify track recording properly stops on device disconnection
Steps:
- Start recording a track
- Disconnect the BLE device
Expected Result:
- Recording automatically stops when device disconnects
- Track data is properly saved
- UI reflects stopped recording state
Success Criteria: Track recording integration works seamlessly with connection state

Maria: Done

### Checklist before requesting a review

[ ] I have performed a self-review of my code
[ ] I have added or updated tests
[ ] I have added or updated relevant documentation

### Additional Information

Screenshot_20250826-100558

Screenshot_20250826-102117

### CODE REVIEW

File: lib/ui/screens/home_screen.dart [hunk 1]
```diff
@@ -26,91 +27,112 @@ class HomeScreen extends StatelessWidget {
...
+        children: [
+          // Error banner with spacing
+          ValueListenableBuilder<bool>(
+            valueListenable: bleBloc.connectionErrorNotifier,
+            builder: (context, error, child) {
+              if (error == true) {
+                return Column(
+                  children: [
+                    const SizedBox(height: 48),
+                    _ConnectionErrorBanner(bleBloc: bleBloc),
```
Comment (mashazyu) [inline lib/ui/screens/home_screen.dart:41]: **suggestion:** I updated error message styling and extracted it to a separate file.

**Review by felixerdy — APPROVED** [review]

**Comment by felixerdy** [comment]
Tests with older version of senseBox:bike was successful. Will merge after #160 is merged
